mesosphere/charts
# kommander: Add centralized kubecost

Adds centralized kubecost on kommander - runs a separate prometheus instance for kubecost and a separate thanos query which is set up to pull kubecost metric data remotely from managed cluster kubecost prometheis. Reuses existing centralized monitoring grafana.

## Outstanding work
- Licensing - but we can add these changes in later
- Alertmanager to monitor and alert on kubecost health - add later, non-blocking

## Testing
I have tested running this version of kommander with centralized kubecost and pointing the KCL controller manager to `dev-thanos` image which I built and pushed using [this branch](https://github.com/mesosphere/kommander-cluster-lifecycle/pull/515). I provisioned a cluster from kommander and was able to verify that the managed cluster's kubecost thanos ELB was getting pulled into the kommander kubecost thanos query stores. I was able to see data getting pulled from the managed cluster using the kommander kubecost thanos query UI. The kommander kubecost UI likewise is installed successfully along with all the other kubecost components.

1. Point kubeaddons-kommander branch to `configVersion: gracedo/test_kubecost`
1. Use `dev-thanos` image for KCL controller manager
     ```
    addonsList:
    - name: kommander
      enabled: true
      values: |
        kommander-cluster-lifecyle:
          controller:
            containers:
              manager:
                image:
                  tag: dev-thanos
    ```
    ^ This actually didn't work for me, so when the cluster came up, I edited the `kcl-cm` deployment to use `dev-thanos` image
1. `konvoy up`
1. [Provision a managed cluster](https://github.com/mesosphere/kommander-cluster-lifecycle#creating-konvoy-clusters)
1. Check thanos query UI for added store from managed cluster: `<CLUSTER_URL>/ops/portal/kommander/kubecost/query/stores`
1. Check Kubecost UI `<CLUSTER_URL>/ops/portal/kommander/kubecost/detail.html`. NOTE: Data won't actually be presented in the UI until approx 2hr after the cluster is joined. The Kubecost team has informed us that as long as we can see the data flowing in to Thanos Query, that should be good enough to confirm. I was able to keep managed clusters up long enough to verify that data shows up in kubecost.
1. Add license key for enterprise version on the Kubecost UI Home page `<CLUSTER_URL>/ops/portal/kommander/kubecost/overview.html`, via `UPGRADE` link on top right (ping me for the key). **licensing integration is TBD at this point, and blocking this release**
![image](https://user-images.githubusercontent.com/5897740/81852279-a136e980-950f-11ea-8d62-ed82c5d1f5e4.png)

1. Kubecost grafana dashboards are installed into the centralized monitoring grafana instance: `<CLUSTER_URL>/ops/portal/kommander/monitoring/grafana` and are also linked to from the Kubecost UI. It is using Thanos query as a datasource so data should already be present in those dashboards.

![image](https://user-images.githubusercontent.com/5897740/81993150-f5b69380-95f9-11ea-8b52-18ec3a57af00.png)
![image](https://user-images.githubusercontent.com/5897740/81993187-049d4600-95fa-11ea-99a2-c384d66d9ce0.png)
